src/base/Assure.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_BASE_ASSURE_H
+#define SQUID_SRC_BASE_ASSURE_H
+
+#include "base/Here.h"
+
+/// Reports the described assertion (at the given debugging level) and throws
+/// the corresponding exception. Reduces compiled code size of Assure() and
+/// Must() callers. Do not call directly; use Assure() instead.
+/// \param description explains the condition (i.e. what MUST happen)
+[[ noreturn ]] void ReportAndThrow_(int debugLevel, const char *description, const SourceLocation &);
+
+/// Calls ReportAndThrow() if needed. Reduces caller code duplication.
+/// Do not call directly; use Assure() instead.
+/// \param description c-string explaining the condition (i.e. what MUST happen)
+#define Assure_(debugLevel, condition, description, location) \
```

Comment:
> but this logic does not work in reverse: Since ReportAndThrow_() is reused by non-~asserting~assuring code, that function should not and cannot be subject to `NDEBUG`.

Without Must() changes in this PR there is no such "non-~asserting~assuring code". The reverse logic works fine.

Reply:
> Without Must() changes ...

Must() changes exist and are discussed in another change request. AFAICT, you tried to make a case that this change request is based on a _self-contained_ logic/principle/truth (rather than dependent on another change request resolution). That did not work. If you actually meant that this change request is simply dependent on a particular resolution of the other change request, then I agree, and (still) consider this change request addressed (because that particular resolution has been rejected).

Please unblock this PR.